### PR TITLE
Add search to filter logs on log table view controller

### PR DESCRIPTION
**Pull request by shawnwelch**
Add UISearchController to ARKLogTableViewController to filter logs based on a search string

### CODE REVIEW

File: Log Viewing/ARKLogTableViewController.m [hunk 1]
```diff
@@ -386,10 +413,21 @@ - (void)_reloadLogs;
...
         [self.tableView reloadData];
     }];
 }
 
+- (void)_reloadFilteredLogs;
+{
+    if (self.searchString.length > 0) {
+        self.filteredLogs = [self.filteredLogs filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"text CONTAINS[c] %@", self.searchString]];
+    } else {
+        self.filteredLogs = [self.logMessages copy];
```
Comment (dfed) [inline Log Viewing/ARKLogTableViewController.m]: no need for `copy` here

File: Log Viewing/ARKLogTableViewController.m [hunk 2]
```diff
@@ -138,6 +141,11 @@ - (void)viewDidLoad;
     self.tableView.rowHeight = 34.0;
     self.tableView.dataSource = self;
     self.tableView.delegate = self;
+    
+    self.searchController = [[UISearchController alloc] initWithSearchResultsController:nil];
```
Comment (dfed) [inline Log Viewing/ARKLogTableViewController.m]: curious if this works if you just `[UISearchController new]`

File: Log Viewing/ARKLogTableViewController.m [hunk 3]
```diff
@@ -31,9 +31,10 @@
 #import "UIActivityViewController+ARKAdditions.h"
 
 
-@interface ARKLogTableViewController () <UIActionSheetDelegate, UIPopoverControllerDelegate>
+@interface ARKLogTableViewController () <UIActionSheetDelegate, UIPopoverControllerDelegate, UISearchResultsUpdating, UISearchControllerDelegate>
```
Comment (dfed) [inline Log Viewing/ARKLogTableViewController.m]: nit: `Controller` before `Results`

**Comment by dfed** [comment]
A few nits. Otherwise looks good!

File: Log Viewing/ARKLogTableViewController.m (hunk 2 above)
Comment (shawnwelch) [inline Log Viewing/ARKLogTableViewController.m]: Nope, you have to use the designated initializer, providing nil means you are planning on displaying the results in the same view that you are searching

**Comment by dfed** [comment]
Just noticed that `UISearchController` is iOS 8 and later. Let's hide this functionality on iOS 6 and 7

**Comment by dfed** [comment]
GORM!
img_9010

**Pull request merged**